h2o-core/src/main/java/water/fvec/C4FChunk.java

```diff
@@ -25,17 +24,36 @@
     return true;
   }
   @Override boolean setNA_impl(int idx) { UnsafeUtils.set4f(_mem,(idx<<2),Float.NaN); return true; }
-  @Override public NewChunk inflate_impl(NewChunk nc) {
-    nc.set_sparseLen(0);
-    nc.set_len(0);
-    final int len = _len;
-    for( int i=0; i<len; i++ ) {
-      float res = UnsafeUtils.get4f(_mem,(i<<2));
-      if( Float.isNaN(res) ) nc.addNum(Double.NaN);
-      else nc.addNum(res);
-    }
+
+  @Override public NewChunk extractRows(NewChunk nc, int from, int to){
+    for(int i = from; i < to; i++)
+      nc.addNum(UnsafeUtils.get4f(_mem,4*i));
+    return nc;
+  }
+  @Override public NewChunk extractRows(NewChunk nc, int... rows){
```

Comment:
Minor note: this is little bit dangerous.

For example:
If i would like to extract first two rows, i can write the following code:
```
extractRows(nc, 0, 1);
```
Expecting that version `extractRows(NewChunk nc, int... rows)` will be called, however in this case
`NewChunk extractRows(NewChunk nc, int from, int to)` is going to be called.